### DISCUSSION TITLE
Hello, i want to compile OBS on Haiku Os, i have installed all the requirements libs

### DISCUSSION THREAD

**Discussion by GatoAmarilloBicolor** [obsproject/obs-studio#4933] [General]
### Operating System Info

Other

### Other OS

Haiku Os (Nightly Build)

### OBS Studio Version

Git

### OBS Studio Log URL

dont have one

### Expected Behavior

I want to know how can compile on haiku os, i have installed all the devel requires as it say for MacOs or Linux but is giving me an error, i want to know how to configure it to compile in my operating system.

### Current Behavior

-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- OBS_VERSION: 25.0.7-254-g0f27cf498-modified
-- Found OpenGL: /boot/system/lib/libGL.so   
-- Found X11: /boot/system/develop/headers   
-- Looking for XOpenDisplay in /boot/system/develop/lib/libX11.so;/boot/system/develop/lib/libXext.so
-- Looking for XOpenDisplay in /boot/system/develop/lib/libX11.so;/boot/system/develop/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - not found
-- Looking for gethostbyname in nsl
-- Looking for gethostbyname in nsl - not found
-- Looking for gethostbyname in bsd
-- Looking for gethostbyname in bsd - not found
-- Looking for connect
-- Looking for connect - not found
-- Looking for connect in socket
-- Looking for connect in socket - not found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - not found
-- Looking for shmat in ipc
-- Looking for shmat in ipc - not found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found FFmpeg: /packages/ffmpeg-4.2.2-9/.self/develop/lib/../lib/libavcodec.so (found version "58.54.100") found components: avcodec avdevice avutil avformat 
CMake Error at /boot/system/data/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Libcurl (missing: CURL_LIB CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /boot/system/data/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindLibcurl.cmake:64 (find_package_handle_standard_args)
  deps/file-updater/CMakeLists.txt:3 (find_package)

-- Configuring incomplete, errors occurred!
See also "/boot/home/src/obs-studio/build/CMakeFiles/CMakeOutput.log".
See also "/boot/home/src/obs-studio/build/CMakeFiles/CMakeError.log".

### Steps to Reproduce

cmake -DUNIX_STRUCTURE=1 -DBUILD_BROWSER=OFF ..

~/src/obs-studio/build> cmake ..
-- OBS_VERSION: 25.0.7-254-g0f27cf498-modified
CMake Error at /boot/system/data/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Libcurl (missing: CURL_LIB CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /boot/system/data/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindLibcurl.cmake:64 (find_package_handle_standard_args)
  deps/file-updater/CMakeLists.txt:3 (find_package)

-- Configuring incomplete, errors occurred!
See also "/boot/home/src/obs-studio/build/CMakeFiles/CMakeOutput.log".
See also "/boot/home/src/obs-studio/build/CMakeFiles/CMakeError.log".

**Comment by WizardCM** [comment]
You're certainly missing dependencies - in this case, libcurl.

**Comment by GatoAmarilloBicolor** [comment]
oh, i didnt know it is called curl_devel for haiku now i am getting that errors on the cmake log, i suppose this will not be easy cause it is a very different Os(posix compatible but not unix-like): 

Determining if the function gethostbyname exists failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_c3691/fast && /bin/make  -f CMakeFiles/cmTC_c3691.dir/build.make CMakeFiles/cmTC_c3691.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_c3691.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname -o CMakeFiles/cmTC_c3691.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_c3691
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_c3691.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname CMakeFiles/cmTC_c3691.dir/CheckFunctionExists.c.o -o cmTC_c3691 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: CMakeFiles/cmTC_c3691.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x10): undefined reference to `gethostbyname'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_c3691.dir/build.make:98: recipe for target 'cmTC_c3691' failed
make[1]: *** [cmTC_c3691] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_c3691/fast' failed
make: *** [cmTC_c3691/fast] Error 2



Determining if the function gethostbyname exists in the nsl failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_642a0/fast && /bin/make  -f CMakeFiles/cmTC_642a0.dir/build.make CMakeFiles/cmTC_642a0.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_642a0.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname -o CMakeFiles/cmTC_642a0.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_642a0
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_642a0.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname CMakeFiles/cmTC_642a0.dir/CheckFunctionExists.c.o -o cmTC_642a0  -lnsl 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lnsl
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_642a0.dir/build.make:98: recipe for target 'cmTC_642a0' failed
make[1]: *** [cmTC_642a0] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_642a0/fast' failed
make: *** [cmTC_642a0/fast] Error 2



Determining if the function gethostbyname exists in the bsd failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_13dec/fast && /bin/make  -f CMakeFiles/cmTC_13dec.dir/build.make CMakeFiles/cmTC_13dec.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_13dec.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname -o CMakeFiles/cmTC_13dec.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_13dec
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_13dec.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=gethostbyname CMakeFiles/cmTC_13dec.dir/CheckFunctionExists.c.o -o cmTC_13dec  -lbsd 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: CMakeFiles/cmTC_13dec.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x10): undefined reference to `gethostbyname'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_13dec.dir/build.make:98: recipe for target 'cmTC_13dec' failed
make[1]: *** [cmTC_13dec] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_13dec/fast' failed
make: *** [cmTC_13dec/fast] Error 2



Determining if the function connect exists failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_14ca9/fast && /bin/make  -f CMakeFiles/cmTC_14ca9.dir/build.make CMakeFiles/cmTC_14ca9.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_14ca9.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=connect -o CMakeFiles/cmTC_14ca9.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_14ca9
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_14ca9.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=connect CMakeFiles/cmTC_14ca9.dir/CheckFunctionExists.c.o -o cmTC_14ca9 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: CMakeFiles/cmTC_14ca9.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x10): undefined reference to `connect'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_14ca9.dir/build.make:98: recipe for target 'cmTC_14ca9' failed
make[1]: *** [cmTC_14ca9] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_14ca9/fast' failed
make: *** [cmTC_14ca9/fast] Error 2



Determining if the function connect exists in the socket failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_49a49/fast && /bin/make  -f CMakeFiles/cmTC_49a49.dir/build.make CMakeFiles/cmTC_49a49.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_49a49.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=connect -o CMakeFiles/cmTC_49a49.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_49a49
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_49a49.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=connect CMakeFiles/cmTC_49a49.dir/CheckFunctionExists.c.o -o cmTC_49a49  -lsocket 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lsocket
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_49a49.dir/build.make:98: recipe for target 'cmTC_49a49' failed
make[1]: *** [cmTC_49a49] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_49a49/fast' failed
make: *** [cmTC_49a49/fast] Error 2



Determining if the function shmat exists failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_79f7e/fast && /bin/make  -f CMakeFiles/cmTC_79f7e.dir/build.make CMakeFiles/cmTC_79f7e.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_79f7e.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=shmat -o CMakeFiles/cmTC_79f7e.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_79f7e
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_79f7e.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=shmat CMakeFiles/cmTC_79f7e.dir/CheckFunctionExists.c.o -o cmTC_79f7e 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: CMakeFiles/cmTC_79f7e.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x10): undefined reference to `shmat'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_79f7e.dir/build.make:98: recipe for target 'cmTC_79f7e' failed
make[1]: *** [cmTC_79f7e] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_79f7e/fast' failed
make: *** [cmTC_79f7e/fast] Error 2



Determining if the function shmat exists in the ipc failed with the following output:
Change Dir: /boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp

Run Build Command(s):/bin/make -f Makefile cmTC_c2d66/fast && /bin/make  -f CMakeFiles/cmTC_c2d66.dir/build.make CMakeFiles/cmTC_c2d66.dir/build
make[1]: Entering directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_c2d66.dir/CheckFunctionExists.c.o
/bin/cc   -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=shmat -o CMakeFiles/cmTC_c2d66.dir/CheckFunctionExists.c.o -c /boot/system/data/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_c2d66
/boot/system/bin/cmake -E cmake_link_script CMakeFiles/cmTC_c2d66.dir/link.txt --verbose=1
/bin/cc -Wall -Wextra -Wvla -Wno-unused-function -Werror-implicit-function-declaration -Wno-missing-braces -Wno-missing-field-initializers  -std=gnu99 -fno-strict-aliasing -DCHECK_FUNCTION_EXISTS=shmat CMakeFiles/cmTC_c2d66.dir/CheckFunctionExists.c.o -o cmTC_c2d66  -lipc 
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/8.3.0/../../../../x86_64-unknown-haiku/bin/ld: cannot find -lipc
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_c2d66.dir/build.make:98: recipe for target 'cmTC_c2d66' failed
make[1]: *** [cmTC_c2d66] Error 1
make[1]: Leaving directory '/boot/home/src/obs-studio/build/CMakeFiles/CMakeTmp'
Makefile:127: recipe for target 'cmTC_c2d66/fast' failed
make: *** [cmTC_c2d66/fast] Error 2

**Comment by catielanier** [comment]
This will likely take a massive port undertaking, simply put, Haiku and Linux kernel APIs are different, so even if it builds it would likely crash.
